### PR TITLE
Docs: Modernize footer design and improve visual hierarchy

### PR DESCRIPTION
**Pull request by SuhaniBh**
### Description
The current Bootstrap footer design, while functional, can benefit from modern design principles to enhance visual appeal and user experience. This PR modernizes the footer with improved spacing, typography, and visual hierarchy while maintaining its core functionality.

### Motivation & Context
This change improves the visual presentation of Bootstrap's documentation footer by:
- Enhancing readability with better spacing and typography
- Implementing a more modern color scheme with subtle backgrounds
- Adding smooth hover interactions for better user feedback
- Improving mobile responsiveness and content organization
- Maintaining accessibility standards while upgrading the design

### Type of changes
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist
- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/github/CONTRIBUTING.md)
- [x] My code follows the code style of the project (using npm run lint)
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### Live previews
- <https://deploy-preview-40969--twbs-bootstrap.netlify.app/>
- Direct link to footer section: https://deploy-preview-40969--twbs-bootstrap.netlify.app/#footer

### Related issues
- N/A - Enhancement proposal for footer design modernization

### CODE REVIEW

File: site/layouts/partials/footer.html [hunk 1]
```diff
@@ -1,59 +1,78 @@
...
+          <li class="mb-3"><a href="https://github.com/sponsors/twbs" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Corporate sponsors</a></li>
+          <li class="mb-3"><a href="{{ .Site.Params.opencollective }}" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Open Collective</a></li>
+          <li class="mb-3"><a href="https://stackoverflow.com/questions/tagged/bootstrap-5" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Stack Overflow</a></li>
         </ul>
       </div>
     </div>
   </div>
 </footer>
+
+<style>
```
Comment (julien-deramond) [inline site/layouts/partials/footer.html:72]: This page can't use `<style>`, the CSS part should be put inside the corresponding (S)CSS file.

File: site/layouts/partials/footer.html [hunk 2]
```diff
@@ -1,59 +1,78 @@
...
-          <li class="mb-2"><a href="https://stackoverflow.com/questions/tagged/bootstrap-5" target="_blank" rel="noopener">Stack Overflow</a></li>
+          <li class="mb-3"><a href="{{ .Site.Params.github_org }}/bootstrap/issues" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Issues</a></li>
+          <li class="mb-3"><a href="{{ .Site.Params.github_org }}/bootstrap/discussions" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Discussions</a></li>
+          <li class="mb-3"><a href="https://github.com/sponsors/twbs" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Corporate sponsors</a></li>
+          <li class="mb-3"><a href="{{ .Site.Params.opencollective }}" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Open Collective</a></li>
+          <li class="mb-3"><a href="https://stackoverflow.com/questions/tagged/bootstrap-5" target="_blank" rel="noopener" class="text-decoration-none text-body-secondary hover-primary">Stack Overflow</a></li>
```
Comment (julien-deramond) [inline site/layouts/partials/footer.html:65]: For the links style, it should be handled in the corresponding (S)CSS file, not by repeating classes all over the file.

File: site/layouts/partials/footer.html [hunk 3]
```diff
@@ -1,59 +1,78 @@
-<footer class="bd-footer py-4 py-md-5 mt-5 bg-body-tertiary">
-  <div class="container py-4 py-md-5 px-4 px-md-3 text-body-secondary">
-    <div class="row">
+<footer class="bd-footer py-5 mt-5 bg-primary-subtle">
+  <div class="container py-5 px-4 px-md-3">
+    <div class="row g-4">
+      <!-- Brand Column -->
```
Comment (julien-deramond) [inline site/layouts/partials/footer.html:4]: The comments added in this PR can be removed.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Thanks for creating the PR, @SuhaniBh!

There are a few code issues that need addressing, but let's hold off on fixing them until we hear back from @mdo (who designs Bootstrap) about the design. We don't want to spend time on code adjustments if the rendering doesn't meet the design expectations.

On a personal note, though I'm not a designer, I find the light version a bit too colorful. I also think the logo and 'Bootstrap' text shouldn't be blue.

**Comment by mdo** [comment]
Thanks, but we'll pass.

**Pull request closed without merging**